File: homework/hardware_software/homework5/51215902152-侯雪城-第五次作业/4.1题源码/Matlab源码/排序源码/bubble_sort.m
function s=bubble_sort(s)
    temp=0;
    len=length(s);
    for i=1:len
        for k=1:len-i
            if s(k)>s(k+1)
                temp=s(k);
                s(k)=s(k+1);
                s(k+1)=temp;
            end
        end
    end
end

File: homework/hardware_software/homework5/51215902152-侯雪城-第五次作业/4.2题源码/排序源码/BubbleSort/bubble_sort.c
#include"bubble_sort.h"

void bubble_sort(uint8 *arr,uint8 len){
	uint8 temp=0;
	for(uint8 i=0;i<len;i++){
		for(uint8 k=0;k<len-i;k++){
			if(arr[k]>arr[k+1]){
				temp=arr[k];
				arr[k]=arr[k+1];
				arr[k+1]=temp;
			}
		}
	}
}
